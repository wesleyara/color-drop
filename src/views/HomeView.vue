<script setup lang="ts">
import HomeRoot from "../components/Frames/HomeRoot.vue";
import HomeTitle from "../components/Frames/HomeTitle.vue";
</script>

<template>
  <HomeRoot>
    <HomeTitle />
  </HomeRoot>
</template>
