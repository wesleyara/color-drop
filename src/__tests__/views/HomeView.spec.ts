/* eslint-disable @typescript-eslint/no-explicit-any */
import { describe, it, expect, beforeEach } from "vitest";
import { mount, VueWrapper } from "@vue/test-utils";
import HomeView from "../../views/HomeView.vue";

describe("HomeView unit tests", async () => {
  let sut: VueWrapper<any>;

  beforeEach(async () => {
    sut = mount(HomeView);
  });

  it("should render the component", () => {
    const wrapper = sut.exists();

    expect(wrapper).toBeDefined();
  });

  it("should match the snapshot", () => {
    const wrapper = sut.html();

    expect(wrapper).toMatchSnapshot();
  });

  it("should have a class", () => {
    const wrapper = sut.classes();

    expect(wrapper).toContain("flex");
  });

  it("should have a main element", () => {
    const wrapper = sut.find("main");

    expect(wrapper.exists()).toBeTruthy();
  });
});
