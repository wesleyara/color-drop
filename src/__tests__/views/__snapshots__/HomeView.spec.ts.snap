// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeView unit tests > should match the snapshot 1`] = `
"<main class="flex flex-col items-center justify-center gap-5">
  <h1 class="text-5xl">Color Drop</h1>
</main>"
`;
